cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -202,38 +202,40 @@ public CruiseControlApiImpl(Vertx vertx, int idleTimeout) {
                                                 json.getString(CC_REST_API_ERROR_KEY)));
                             }
                             JsonObject statusJson = new JsonObject();
-                            String taskStatus = jsonUserTask.getString("Status");
-                            statusJson.put("Status", taskStatus);
-                            // The status could be ACTIVE in which case there will not be a "summary" so we check that we are
-                            // in a state that actually has that key.
-                            if (taskStatus.equals(CruiseControlUserTaskStatus.IN_EXECUTION.toString()) ||
-                                    taskStatus.equals(CruiseControlUserTaskStatus.COMPLETED.toString())) {
-                                // We now need to extract the original response which is in a raw string (not nicely formatted JSON)
-                                statusJson.put("summary", ((JsonObject) Json.decodeValue(jsonUserTask.getString("originalResponse"))).getJsonObject("summary"));
+                            String taskStatusStr = jsonUserTask.getString("Status");
+                            statusJson.put("Status", taskStatusStr);
+                            CruiseControlUserTaskStatus taskStatus = CruiseControlUserTaskStatus.lookup(taskStatusStr);
+                            switch (taskStatus) {
+                                case ACTIVE:
+                                    // If the status is ACTIVE there will not be a "summary" so we skip pulling the summary key
+                                    break;
+                                case IN_EXECUTION:
+                                    // Tasks in execution will be rebalance tasks, so their original response will contain the summary of the rebalance they are executing
+                                    // We handle these in the same way as COMPLETED tasks so we drop down to that case.
+                                case COMPLETED:
+                                    // Completed tasks will have the original rebalance proposal summary in their original response
+                                    statusJson.put("summary", ((JsonObject) Json.decodeValue(jsonUserTask.getString("originalResponse"))).getJsonObject("summary"));
+                                    break;
+                                case COMPLETED_WITH_ERROR:
+                                    // Completed with error tasks will have "CompletedWithError" as their original response, which is not Json.
+                                    statusJson.put("summary", jsonUserTask.getString("originalResponse"));
```

Comment:
How is it reflected in the resource status? The user will see the "summary" with the original proposal and the final state will be Ready but no evidence that it's completed with error?

Reply:
No the status of the field is still `CompletedWithError` and this will be picked up in the KRAO `onRebalancing` method  where the status will be set to `NotReady`.  We could just have a blank `summary` but in future this will hopefully contain details of the actual error (I have a rough idea for upstream CC PR to retain that info for the user task).

Reply:
> No the status of the field is still CompletedWithError

which status field? in the next sentence you said it's NotReady.

> and this will be picked up in the KRAO onRebalancing method where the status will be set to NotReady

so you are saying that the `KafkaRebalance` will be in `NotReady` status but with a valid summary? I think it's really confusing for the user.

Reply:
This is the CC client? It doesn't touch the `KafkaReabalance` status field. The KRAO does that in the [onRebalancing](https://github.com/tomncooper/strimzi-kafka-operator/blob/3ccfe2d3e4243e0b53b4e0c93b8c9205bcc97be3/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java#L687) method. The _status_ field of the `CruiseControlReponse`'s json is `CompletedWithError` which will trigger the appropriate branch in the KRAO which will not set a summary (which matches what it does currently).

Reply:
You are right! I did too many reviews today! ;-)